server/src-lib/Hasura/SQL/Types.hs

```diff
@@ -520,3 +526,49 @@ instance (ToSQL a) => ToSQL (PGType a) where
     PGTypeScalar ty -> toSQL ty
     -- typename array is an sql standard way of declaring types
     PGTypeArray ty -> toSQL ty <> " array"
+
+data PGTypeKind
+  = PGKindBASE
+  | PGKindCOMPOSITE
+  | PGKindDOMAIN
+  | PGKindENUM
+  | PGKindRANGE
+  | PGKindPSEUDO
```

Comment:
I think these should still use the usual naming convention, e.g. `PGKindBase`, `PGKindComposite`, etc., with a custom `ToJSON` instance if necessary. I also think the `ToJSON` and `FromJSON` instances should roundtrip, but currently they do not.